ndsl/dsl/ndsl_runtime.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations
+
+import inspect
+import warnings
+from typing import Any, Callable
```

Comment:
`typing.Callable` is a deprecated alias for `collections.abc.Callable` since python 3.9. Let's try to avoid it. In PR #270 I've updated all usage. I haven't found a way to enforce it by CI yet, so we'll have to catch it in code reviews until then.

```suggestion
from collections.abc import Callable
from typing import Any
```

Reply:
Deep inside of me, I knew it... I was lazy to look around for the correct thing.